docs/design-docs/02-custom-root-reconciler-clusterrole.md

```diff
@@ -18,65 +24,130 @@ This proposal introduces a mechanism to customize this behavior.
 If one wants to allow deploying resources from a source where one doesn't have full
 control of the contents of the upstream config source, and want to limit what another
 team or third party can accomplish in the cluster, one can imagine limiting what
-permissions are granted to the `ServiceAccount` running the reconciler.
+permissions are granted to the `ServiceAccount` running the `RootSync` reconciler.
 
 However, as Config Sync is currently granting `cluster-admin`, any custom role bindings
-are effectively ignored; the reconciler will, due to the binding added by Config Sync,
+are effectively ignored; the `RootSync` reconciler will, due to the binding added by Config Sync,
 have access to do _anything_ regardless.
 
 To follow the [principle of least privilege], one should ensure the reconciler only has
 access to deploy the expected resources.
 
+Additionally, the `RepoSync` API currently requires for users to manage the lifecycle
+of `RoleBinding` objects themselves. The proposed API here is intended to also
+simplify the management of `RoleBinding`s for `RepoSync`s.
+
 See also [#935].
 
 [principle of least privilege]: https://en.wikipedia.org/wiki/Principle_of_least_privilege
 [#935]: https://github.com/GoogleContainerTools/kpt-config-sync/issues/935
 
 ## Design Overview
 
-By providing the name of a user-defined `ClusterRole` for a `RootSync`, a user can
+### `RootSync`s
+
+By providing the name of a list user-defined `RoleRef`s for a `RootSync`, a user can
 override which role Config Sync binds to. This configuration is exposed as a new field
 on `spec.overrides` for a `RootSync`:
 
 ```yaml
+kind: RootSync
+metadata:
+  name: my-root-sync
+  namespace: config-management-system
 spec:
   overrides:
-    clusterRole:
-      name: my-custom-role
+    roleRefs:
+    - kind: ClusterRole # Creates a ClusterRoleBinding bound to ClusterRole my-cluster-role
+      name: my-cluster-role
+    - kind: ClusterRole # Creates a RoleBinding in my-tenant-namespace bound to ClusterRole tenant-cluster-role
+      name: tenant-cluster-role
+      namespace: my-tenant-namespace
+    - kind: Role # Creates a RoleBinding in my-tenant-namespace bound to Role my-tenant-role
+      name: my-tenant-role
+      namespace: my-tenant-namespace
 ```
 
-This role is defaulted to `cluster-admin` in order to stay backwards compatible.
+For `RootSync` objects, the default behavior will create a ClusterRoleBinding
```

Comment:
Exactly how will the defaulting work here? Is the assumption that is no RoleRefs are specified for a RootSync, the binding to cluster-admin will automatically be set up, but if the RoleRefs list is not empty, only the specified bindings will be created?

Reply:
Yeah that sums it up pretty accurately. Keep in mind this is necessary for backwards compatibility - I may not have chosen this behavior if designing from scratch.

Reply:
Suggest: 
> For `RootSync` objects, a default `cluster-admin` ClusterRoleBinding will be applied when `spec.override.roleRefs` is empty or nil, for reverse compatibility.

Reply:
Done